internal/server/singleprocess/state/job.go

```diff
@@ -252,6 +252,49 @@ func (s *State) JobCreate(jobs ...*pb.Job) error {
 	return err
 }
 
+// Given a Project Ref and a Job Template, this function will generate a slice
+// of QueueJobRequests for every application inside the requested Project
+func (s *State) JobProjectScopedRequest(
+	pRef *pb.Ref_Project,
+	jobTemplate *pb.Job,
+) ([]*pb.QueueJobRequest, error) {
+	var result []*pb.QueueJobRequest
+	project, err := s.ProjectGet(pRef)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, app := range project.Applications {
+		// We copy by all values because copying the original job includes copying a mutux lock,
```

Comment:
nit: we don't seem to be copying _all_ the `pb.Job` values here. If we're intentionally only copying a few, we should document that we're only copying the ones we really need. Otherwise someone in the future may read this and think that we copied all of them at the time, but now (then?) there are more, and we're now (then) are not copying them all but should be? 

If we do need to copy all without the mutex issues, we're already using [`copystructure`](https://github.com/mitchellh/copystructure) in a few other places, I believe that would avoid the issue.

Reply:
Ah, I may have mis-read the code comment here. I read it first as "we copy all" but it's actually "we copy by all values" which threw me. That said, do we need to copy all of them? We seem to be omitting `ExpireTime` and `QueueTime` for instance, which sound relevant?

Reply:
Yeah, I might of missed those. Since this is supposed to be more of a generic job func, I think using something like copystructure is the way to go here 👍🏻 I'll update to use that.